ui/app/utils/properties/glimmer-style-string.js

```diff
@@ -0,0 +1,24 @@
+import { htmlSafe } from '@ember/template';
+
+export default function styleString(target, name, descriptor) {
+  if (!descriptor.get) throw new Error('styleString only works on getters');
+  const orig = descriptor.get;
+  descriptor.get = function() {
+    const styles = orig.apply(this);
+
+    let str = '';
+
+    if (styles) {
+      str = Object.keys(styles)
+        .reduce(function(arr, key) {
+          const val = styles[key];
+          arr.push(key + ':' + (typeof val === 'number' ? val.toFixed(2) + 'px' : val));
+          return arr;
+        }, [])
+        .join(';');
+    }
+
+    return htmlSafe(str);
+  };
+  return descriptor;
+}
```

Comment:
Thanks for adding this, I couldn’t get the older implementation working in `DasRecommendationCardComponent` and just used `htmlSafe` instead but this will be nicer!

Reply:
Yeah! I still like the computed/decorator pattern for things like this. I know we talked in the UI guild at one point about maybe not using custom decorators since they are still Stage 2 in tc39 land, but I couldn't help myself.